### PR TITLE
simplify struct's introduction, add to about.md

### LINKED ISSUE #1079: [rust][concept cleanup] simplify structs concept introduction
https://github.com/exercism/v3/blob/master/languages/rust/concepts/structs/introduction.md

A concept introduction should be short, just a few sentences.

### PR DESCRIPTION
**Pull request by efx**
Close #1079.

I'm leaning towards keeping code samples out of our introduction.md's
and into `about.md`.

I like the style of introducing the concept in 1 or 2 lines of prose.

### CODE REVIEW

File: concepts/structs/about.md [hunk 1]
```diff
@@ -1,20 +1,52 @@
...
 - Every field of a struct has a particular name.
 - Visibility is tracked at the field level: each field may be independently public, private, or other[^1].
 
-# Tuples etc.
+
+# Structs Overview
+
+Structs come in three flavors: structs with named fields, tuple structs, and unit structs. 
+
+Structs are defined using the `struct` keyword, followed by the capitalized name of the type the struct is describing:
```
Comment (coriolinus) [inline concepts/structs/about.md:15]:
```suggestion
Structs are defined using the `struct` keyword, followed by the name of the type the struct is describing:
```

Capitalization there is a convention which generates a warning if violated, but it's not intrinsically part of the language.

File: concepts/structs/about.md [hunk 2]
```diff
@@ -1,20 +1,52 @@
...
 - Every struct type has a particular name.
 - Every member of a struct is called a **field**.
 - Every field of a struct has a particular name.
 - Visibility is tracked at the field level: each field may be independently public, private, or other[^1].
 
-# Tuples etc.
+
+# Structs Overview
+
+Structs come in three flavors: structs with named fields, tuple structs, and unit structs.
```
Comment (coriolinus) [inline concepts/structs/about.md:13]: Suggest moving this line into the "Struct Flavors" section.

File: concepts/structs/about.md (hunk 1 above)
Comment (coriolinus) [reply concepts/structs/about.md:15]:
```suggestion
Structs are defined using the `struct` keyword, followed by the capitalized name of the type the struct is describing. The name may then be followed by curly braces (for a standard struct), parentheses (for a tuple struct), or a semicolon (for a unit struct). 
```

Let's express the precise grammar.

File: concepts/structs/about.md (hunk 2 above)
Comment (coriolinus) [reply concepts/structs/about.md:13]:
```suggestion
Structs come in three flavors: structs with named fields (standard structs), tuple structs, and unit structs. 
```

**Review by coriolinus — APPROVED** [review]
Some quibbles, but overall looks good. Be careful with the suggestions; I suspect that just hitting the "apply" button on all of them will result in at least some changes being undone.

**Comment by efx** [comment]
> Be careful with the suggestions; I suspect that just hitting the "apply" button on all of them will result in at least some changes being undone.

Thanks for the tip! You can apparently apply them in one go from the files view using "Add suggestion to batch".
I did the changes locally though for ease.

**Pull request merged**